tensor/interpolateHypercube.cc

```diff
@@ -76,9 +76,16 @@ namespace civita
             auto dst=find_if(targetHC.begin(), targetHC.end(),
                              [&](const XVector& i){return i.name==src.name;});
             if (dst==targetHC.end())
-              throw runtime_error("no matching dimension found");
-            interimHC.xvectors.push_back(*dst);
-            rotation[dst-targetHC.begin()]=i;
+            {
+              // possible alternative when targetHC has no xvectors or undefined ones. for feature 147
+              interimHC.xvectors.push_back(src);
+              rotation[i]=i;
+            }
```

Comment:
I would be OK with this, provided you can guarantee that rotation is a permutation (ie it is filled with the numbers 0..rotation.size(), each appearing only once in the sequence, but not necessarily in order).

I don't think the current code guarantees that, as your chosen position for the ith vector may get overwritten later on on line 87, leading to one of the spots in the vector being uninitialised, or left at its default value of 0.

One solution might be to check that the xvector dimension name appearing as targetHC[i] does not appear in the src, and to choose another index if it is found to be. This is hardly efficient or optimal though - perhaps you have a special use case that might be handled by a different code branch.

Reply:
> I don't think the current code guarantees that, as your chosen position for the ith vector may get overwritten later on on line 87, leading to one of the spots in the vector being uninitialised, or left at its default value of 0.

Ah, I see that now too.

> perhaps you have a special use case that might be handled by a different code branch.

Well, any time one builds a model in which the variables have tensor parameters as initial values this will occur.

I tried to change the dimensions in Edit-> Dimensions after checking that they were mismatched in "dimensions" when browsing the object, but to no avail. If it is possible, it is not simple and the user will likely struggle like I did. Then I also tried to format my CSV file such that the dimensions are (0.000000, 1.000000, 2.000000, etc.,) as when a parameter is initialised using a zeros(5), iota (5),eye(5), etc. But the dimensions were still mismatched and my model wouldn't run.

I might have to go with the solution you suggest, which I will work on now. Yesterday, I tried to improve the other PR.